packages/wp-source/src/actions.ts

```diff
@@ -272,9 +272,32 @@ const actions: WpSource["actions"]["source"] = {
     } = state.source;
 
     if (homepage) {
-      const pattern = `RegExp:^${concatLink(
-        subdirectory
-      )}(?!(\\?([^&#]+&)*s=))`;
+      const regex =
+        // prettier-ignore
+        // This regex is using negative lookahead. It means that the whole regex
+        // matches if the group using the negative lookahead DOES NOT match.
+        "(?!" +         // negative lookahead       
+          "(" +           // group    
+            "\\?" +         // match "?" literally
+            "(" +           // group     
+              "[^&#]+" +      // match any character that is not `&` or `#` 1 or more times 
+              "&" +           // match `&` literally
+            ")*" +          // match the group zero or more times
+            "s=[^&#]*" +    // match a search query param (e.g. `s=something`)
+          ")" +           // end group
+        ")" +           // end negative lookahead
+        "(" +           // group 
+          "\\?" +         // match "?" literally
+          "([^#]+)*" +    // match the query string part of the URL zero or more times
+        ")?" +          // match the group zero or one times
+        "(#.*)?" +      // match the hash part of the URL 
+        "$"; // match line end
+
+      const pattern = `RegExp:^${concatLink(subdirectory)}${regex}`;
+
+      // For those who are able to read that sort of thing the full pattern is:
+      // RegExp:^/(?!(\?([^&#]+&)*s=[^&#]*))(\?([^#]+)*)?(#.*)?$
+      // (It has to be double-escaped in the string form above though)
```

Comment:
Okay, this just blowed my mind, hahaha :exploding_head:

Writing comments inside regexps is actually a good idea, so I did a quick search, and I found [this article](https://keleshev.com/verbose-regular-expressions-in-javascript) explaining how this could be implemented for JavaScript using template literals. I've modified the code a bit –just to allow interpolation– and, voilà, I was able to write something like this: :smile:

```ts
    if (homepage) {
      const homePath = concatLink(subdirectory);
      const regExp = verboseRegExp`
        ^                // Beginning of line.
        ${homePath}      // Path to the homepage.
        (                // Must be followed by:
          $|             // 1. End of line or...
          #|             // 2. Hashtag or...
          \?             // 3. Query...
          (?!            //    not containing
            ([^&#]+&)*   //    ...any parameter (0..n)
            s=           //    ...and the search param (s=)
          )
        )
      `;

      const pattern = `RegExp:${regExp}`;
```

Reply:
BTW, this is the function I wrote:

```ts
/**
 * Write verbose RegExp strings. It supports:
 * - Expressions in multiple lines.
 * - Single-line and multiple-line comments.
 * - Interpolation.
 *
 * @remarks Purely based on
 * https://keleshev.com/verbose-regular-expressions-in-javascript
 *
 * @param input - Input.
 * @param args - Interpolated args.
 * @returns Regexp in string format.
 */
export const verboseRegExp = (
  input: TemplateStringsArray,
  ...args: string[]
) => {
  const source = input.raw.reduce(
    (source, input, index) =>
      source.concat(input, String.raw`${args[index] || ""}`),
    ""
  );

  const regexp = /(?<!\\)\s|[/][/].*|[/][*][\s\S]*[*][/]/g;
  const result = source.replace(regexp, "");

  return result;
};
```

Reply:
Awesome, thanks for digging into this @DAreRodz 🙂 I've seen a similar approach in https://stackoverflow.com/a/53925033/2638310 where I got the idea for concatenating the regex to make it more readable.

> I've modified the code a bit –just to allow interpolation– and, voilà, I was able to write something like this: 😄

Did you mean that you modified the code in this PR (because you didn't push any changes) or were you just testing locally?

Reply:
> Did you mean that you modified the code in this PR (because you didn't push any changes) or were you just testing locally?

I meant [this code](https://gist.github.com/keleshev/c49465caed1f114b2bb3f2b730e221ca) (from the [article](https://keleshev.com/verbose-regular-expressions-in-javascript) I mentioned). :smile: 

An yes, I just tested it locally. I don't think it would be good idea to add a new function that we would have to test and maintain, just to be able to write RegExps that way. But I thought it was worth sharing it anyway. :grin:

Reply:
On the other hand, we could use that function to comment other regexps we have, like this one:

https://github.com/frontity/frontity/blob/ad55f6d854e59d2d2939e03f14de9eca690aa25a/packages/wp-source/src/libraries/route-utils.ts#L105

It could be a good idea in the end. :thinking: